tests/contrib/test_spotify.py

```diff
@@ -0,0 +1,78 @@
+from __future__ import unicode_literals
+
+import pytest
+import responses
+from urlobject import URLObject
+from flask import Flask
+from flask_dance.contrib.spotify import make_spotify_blueprint, spotify
+from flask_dance.consumer import OAuth2ConsumerBlueprint
+from flask_dance.consumer.backend import MemoryBackend
+
+
+def test_blueprint_factory():
+    spotify_bp = make_spotify_blueprint(
+        client_id="foo",
+        client_secret="bar",
+        scope="user-read-private",
+        redirect_to="index",
+    )
+    assert isinstance(spotify_bp, OAuth2ConsumerBlueprint)
+    assert spotify_bp.session.scope == "user-read-private"
+    assert spotify_bp.session.base_url == "https://api.spotify.com"
+    assert spotify_bp.session.client_id == "foo"
+    assert spotify_bp.client_secret == "bar"
+    assert spotify_bp.authorization_url == "https://accounts.spotify.com/authorize"
+    assert spotify_bp.token_url == "https://accounts.spotify.com/api/token"
+
+
+def test_load_from_config():
+    app = Flask(__name__)
+    app.secret_key = "anything"
+    app.config["SPOTIFY_OAUTH_CLIENT_ID"] = "foo"
+    app.config["SPOTIFY_OAUTH_CLIENT_SECRET"] = "bar"
+    spotify_bp = make_spotify_blueprint(redirect_to="index")
+    app.register_blueprint(spotify_bp)
+
+    resp = app.test_client().get("/spotify")
+    url = resp.headers["Location"]
+    client_id = URLObject(url).query.dict.get("client_id")
+    assert client_id == "foo"
+
+
+@responses.activate
+def test_context_local():
+    responses.add(responses.GET, "https://google.com")
+
+    # set up two apps with two different set of auth tokens
+    app1 = Flask(__name__)
+    ghbp1 = make_spotify_blueprint(
```

Comment:
This is totally just a minor nitpick, but I'll mention it anyway. Looks like you copied these tests from the GitHub blueprint: `ghbp1` stands for "github blueprint 1". Could you modify this variable to `sbp1` (for "spotify blueprint 1") instead? It makes the tests a tiny bit more readable.

Reply:
Totally agree with you here on improving readability. Updated the variable names. Good catch! 👍